packages/quantic/cypress/fixtures/folded-result-list/exampleResultWithMoreChildResults.json

```diff
@@ -67,11 +70,11 @@
       "title": "Feather Stars",
       "children": [
         {"title": "Antarctic Feather Star"},
+        {"title": "Klunzinger's Feather Star"},
```

Comment:
The reason why the results were returned in that order before is that the array returned by the previous version of the `removeDuplicates` function was generated by running [`Object.values`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values) on an object.

`Object.values` iterates over non-negative integer keys in ascending order by value (in that regard, it behaves as a `for..in` loop, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#description). Since we're filtering out duplicates based on the `uniqueId`, and the `uniqueId` values in the example results are numbers, those numbers became keys in the `removeDuplicates` reduce function, and were then returned in ascending order by `Object.values`.

The new `removeDuplicates` method uses a `Map` (instead of `Record`) when calling `reduce`, and relies on [`Map.prototype.values()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/values) for generating the return array, so that the initial order is preserved even if the keys are numbers.